app/src/main/java/com/fstyle/structure_android/data/source/local/realm/RealmApi.java

```diff
@@ -9,10 +9,41 @@
  * Created by le.quang.dao on 14/03/2017.
  */
 
-public interface RealmApi {
-    <T> Observable<T> realmTransaction(Action2<Subscriber<? super T>, Realm> action);
+public class RealmApi {
 
-    <T> Observable<T> realmOnNewThread(Action2<Subscriber<? super T>, Realm> action);
+    private Realm mRealm;
 
-    void closeRealmOnMainThread();
+    public RealmApi() {
+        mRealm = Realm.getDefaultInstance();
+    }
+
+    /**
+     * <h1>USE THIS METHOD FOR <b>INSERT/UPDATE/DELETE (On Background Thread)</b></h1><br>
+     * Use this method for insert/delete action on action.
+     * <br><b>BIG PROBLEM</b>: be careful with this method. action object cannot use across thread.
+     * this
+     * method
+     * execute a async transaction so, if you let subscribers onNext with action object. Realm
+     * object will become invalid and cause bug.
+     */
+    public <T> Observable<T> realmTransactionAsync(
+            final Action2<Subscriber<? super T>, Realm> action) {
+        return Observable.create(subscriber -> mRealm.executeTransactionAsync(
+                realm -> action.call(subscriber, realm), subscriber::onCompleted,
```

Comment:
ERROR: 'lambda arguments' have incorrect indentation level 16, expected level should be 12.